components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -248,6 +250,14 @@ protected void triggerNotification(User user, String type, String code, Property
     protected User getUser(Map eventProperties, UserStoreManager userStoreManager){
 
         String userName = (String) eventProperties.get(IdentityEventConstants.EventProperty.USER_NAME);
+        try {
+            if(!userStoreManager.isExistingUser(userName)){
```

Comment:
What is the scenario, a non existing user triggering this flow and get to this point?

As I see, this handler registered to PRE_ADD_USER, POST_ADD_USER events, and those are getting triggered only for the user store that user exists.